hypertrace-core-graphql-attribute-store/src/main/java/org/hypertrace/core/graphql/attributes/AttributeModelTranslator.java

```diff
@@ -66,40 +106,36 @@ public AttributeKind convertType(AttributeModelType type) {
                     String.format("Unrecognized attribute type %s", type.name())));
   }
 
-  private List<AttributeModelMetricAggregationType> convertMetricAggregationTypes(
+  private List<AttributeModelMetricAggregationType> convertMetricAggregationFunctions(
       List<AggregateFunction> aggregationTypes) {
     return aggregationTypes.stream()
         .map(this::convertMetricAggregationType)
         .collect(Collectors.toUnmodifiableList());
   }
 
+  private List<AggregateFunction> convertMetricAggregationTypes(
+      List<AttributeModelMetricAggregationType> aggregationTypes) {
+    return aggregationTypes.stream()
+        .map(this::convertMetricAggregationType)
+        .collect(Collectors.toUnmodifiableList());
+  }
+
   private AttributeModelMetricAggregationType convertMetricAggregationType(
       AggregateFunction aggregateFunction) {
-    switch (aggregateFunction) {
-      case COUNT:
-        return AttributeModelMetricAggregationType.COUNT;
-      case AVG:
-        return AttributeModelMetricAggregationType.AVG;
-      case SUM:
-        return AttributeModelMetricAggregationType.SUM;
-      case MIN:
-        return AttributeModelMetricAggregationType.MIN;
-      case MAX:
-        return AttributeModelMetricAggregationType.MAX;
-      case AVGRATE:
-        return AttributeModelMetricAggregationType.AVGRATE;
-      case PERCENTILE:
-        return AttributeModelMetricAggregationType.PERCENTILE;
-      case DISTINCT_COUNT:
-        return AttributeModelMetricAggregationType.DISTINCT_COUNT;
-      case DISTINCT_ARRAY:
-        return AttributeModelMetricAggregationType.DISTINCT_ARRAY;
-      case AGG_UNDEFINED:
-      case UNRECOGNIZED:
-      default:
-        throw new UnknownFormatConversionException(
-            String.format("Unrecognized aggregate function %s", aggregateFunction.name()));
-    }
+    return Optional.ofNullable(AGGREGATION_TYPE_MAPPING.get(aggregateFunction))
+        .orElseThrow(
+            () ->
+                new UnknownFormatConversionException(
```

Comment:
nit - pre-existing older code, but now we want to try to always throw status exceptions

Reply:
Reverted all the changes in this file.